### ISSUE TITLE
creating smiles strings in rdkit python dumps core

### ISSUE THREAD

**Issue by drmatthewclark** [rdkit/rdkit#3004]
**Configuration:**

- RDKit Version:
- Operating system:
- Python version (if relevant):
- Are you using conda?
- If you are using conda, which channel did you install the rdkit from?
- If you are not using conda: how did you install the RDKit?

(conda-env) [mclark@shemini rmc]$ uname -a
Linux shemini 5.5.7-200.fc31.x86_64 #1 SMP Fri Feb 28 17:18:37 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

(conda-env) [mclark@shemini rmc]$ python --version
Python 3.6.10 :: Anaconda, Inc.

python[70853]: segfault at 0 ip 00007fffda9f8cee sp 00007fffffffb620 error 4 in libRDKitFileParsers.so.1.2017.09.1[7fffda9b7000+c7000]
kernel: Code: 8b 74 24 28 45 31 c0 48 8b 7c 24 18 31 c9 e8 79 34 fe ff 48 81 c4 c8 03 00 00 5b 5d 41 5c 41 5d 41 5e 41 5f c3 48 8b 54 24 10 <48> 8b 02 48 89 d>

```
Your code sample here
```   try:
        mol = Chem.MolFromMolBlock(sdfile)
        if mol:
           smiles = Chem.MolToSmiles(mol, isomericSmiles=True,canonical=True)
           tags['smiles'] = smiles
        else:
           mol = ''
    except:
        mol = ''



<!-- If you have a support question, please use the rdkit-discuss mailing list: 
https://sourceforge.net/p/rdkit/mailman/ -->

**Comment by greglandrum** [comment]
Hi @drmatthewclark : this is clearly something that shouldn't happen. Is it possible for you to share the SD file so that I can try and track down the problem?

**Comment by drmatthewclark** [comment]
[badrmc.txt](https://github.com/rdkit/rdkit/files/4338252/badrmc.txt)
This file appears to cause core dump.  It may be incorrect but should  not cause a core dump.

**Comment by greglandrum** [comment]
I can't reproduce the crash with either the current released version of the RDKit (2019.09.3) or master.
Which operating system are you on and which RDKit version are you using?

**Comment by drmatthewclark** [comment]
The os and rdkit is in the report.  The crash occurs on that molecule that is attached, but it is about the 60,000th molecule in the file so previous molecules may have corrupted something.

**Comment by d-b-w** [comment]
It looks like the RDKit version is missing from the report. You can include find the RDKit version like:

    python -c "import rdkit; print(rdkit.__version__)"

**Comment by drmatthewclark** [comment]
version:  (conda-env) [mclark@shemini loader]$ python -c "import rdkit; print(rdkit.__version__)"
2017.09.1
(conda-env) [mclark@shemini loader]$

**Comment by greglandrum** [comment]
There certainly are conda builds available for 2019.09. If `conda update rdkit` doesn’t work for you, could you please share the results of `conda list | grep rdkit` so that we can see where the install is from?

**Comment by drmatthewclark** [comment]
(conda-env) [mclark@shemini loader]$ conda list | grep rdkit
boost                     1.63.0           py36h415b752_1    rdkit
rdkit                     2017.09.1                py36_1    rdkit
rdkit-postgresql          2019.09.3.0          h5fa6086_0    rdkit

btw I also have an issue I am trying to isolate where queries with the cartridge cause the postgresql database to crash using the @= operator.
